From 5102f0771f42aa95c97a5e32bd5cdee02008b625 Mon Sep 17 00:00:00 2001
From: Prasanna S <prasannas@ami.com>
Date: Tue, 19 Aug 2025 09:28:35 +0530
Subject: [PATCH 2/2] Fix for BackupImage not showing in WebUI

Ensure the bmc_bkup D-Bus entry is generated based on the
non-functional image located in the media directory.

Upstream-Status: Inappropriate

Signed-off-by: Prasanna S <prasannas@ami.com>
Signed-off-by: Meghana G <meghanag@ami.com>
Signed-off-by: Mohammed Javith Akthar M <mohammedjavitham@ami.com>
---
 bmc/item_updater.cpp | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/bmc/item_updater.cpp b/bmc/item_updater.cpp
index a0cc2954..acddd311 100644
--- a/bmc/item_updater.cpp
+++ b/bmc/item_updater.cpp
@@ -273,7 +273,7 @@ void ItemUpdater::processBMCImage()
         if (0 ==
             iter.path().native().compare(0, BMC_RO_PREFIX_LEN, BMC_ROFS_PREFIX))
         {
-            std::string id = "bmc_active";
+            std::string id ;
             // upstream changed this to relative_path ... is that right?
             fs::path releaseFile(OS_RELEASE_FILE);
             auto osRelease = iter.path() / releaseFile.relative_path();
@@ -328,9 +328,14 @@ void ItemUpdater::processBMCImage()
             {
                 // Set functional to true and remove the functional suffix
                 functional = true;
+                id="bmc_active";
                 flashId.erase(flashId.length() - strlen(functionalSuffix));
                 functionalFound = true;
             }
+            else
+            {
+                id="bmc_bkup";
+            }

             auto purpose = server::Version::VersionPurpose::BMC;
             restorePurpose(flashId, purpose);
-- 
2.43.0

